src/common/transformations/src/transformations/common_optimizations/transpose_sinking_unary.cpp

```diff
@@ -0,0 +1,106 @@
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "openvino/util/log.hpp"
+#include "itt.hpp"
+#include <ngraph/opsets/opset9.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "transformations/common_optimizations/transpose_sinking_unary.hpp"
+
+#include <utility>
+
+namespace {
+
+using NodePtr = std::shared_ptr<ov::Node>;
+using NodePair = std::pair<NodePtr, NodePtr>;
+
+NodePair SwapNodes(NodePtr first_node, NodePtr second_node) {
+    auto second_node_inputs = second_node->input_values();
+    second_node_inputs[0] = first_node->input_value(0);
```

Comment:
Not sure that this is the right approach for replacing inputs for ports. I think you should use `replace` method of `Output<Node>` instead.
@itikhono, please advice

Reply:
I think this is correct but this approach causes re-creation of the nodes. We can avoid it if we will use `replace` method.